lib/vertica/query.rb

```diff
@@ -18,7 +18,8 @@ class Vertica::Query
   #   to receive data for a `COPY ... FROM STDIN` statement.
   def initialize(connection, sql, row_handler: nil, copy_handler: nil)
     @connection, @sql = connection, sql
-    @buffer = row_handler.nil? && copy_handler.nil? ? [] : nil
+    @reset_buffer_handler = lambda { row_handler.nil? && copy_handler.nil? ? [] : nil }
```

Comment:
To do it, I have to store the original row_handler, and copy_handler although @raw_handler would be a lambda created in the constructor. Do you prefer it?